pkg/provision/ec2.go

```diff
@@ -61,6 +65,15 @@ func (p *EC2Provisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
 		},
 	})
 	if err != nil {
+		// clean up SG if there was an issue provisioning the EC2 instance
```

Comment:
Before this change, several dozen security groups got created (because of retries?) and orphaned via inlets-operator when the ec2 provision call failed.

Reply:
:+1: good spot, thanks